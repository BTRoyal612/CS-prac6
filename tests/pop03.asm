// The machine code generated by your program should go here
// pop pointer0
@0
D=A
@3
D=A+D
@R13
M=D
@SP
AM=M-1
D=M
@R13
A=M
M=D

// pop temp0
@0
D=A
@5
D=A+D
@R13
M=D
@SP
AM=M-1
D=M
@R13
A=M
M=D

(END)
@END
0;JMP