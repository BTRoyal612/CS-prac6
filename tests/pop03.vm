// pop temp and pointer
pop pointer 0
pop temp 0