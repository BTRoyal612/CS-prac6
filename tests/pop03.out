| RAM[0] | RAM[1] | RAM[2] | RAM[3] | RAM[4] | RAM[5] |RAM[256]|RAM[257]|RAM[258]|RAM[300]|RAM[301]|RAM[400]|RAM[401]|RAM[3000|RAM[3001|RAM[3010|RAM[3011|
|    256 |    300 |    400 |   -200 |   3010 |    100 |    100 |   -200 |      0 |      0 |      0 |      0 |      0 |      0 |      0 |      0 |      0 |
